File: C++/120. Triangle.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    int minimumTotal(std::vector<std::vector<int>>& triangle) {
        auto n = int(triangle.size());
        std::vector<int> res (triangle[n - 1].begin(), triangle[n - 1].end());
        for (int i = n - 1; i > 0; i--) {
            for (int j = 0; j < i; j++) {
                res[j] = std::min(res[j], res[j + 1]) + triangle[i - 1][j];
            }
        }
        return res[0];
    }
};

File: Go/120. Triangle.go
package main

func minimumTotal(triangle [][]int) int {
	if len(triangle) == 0 {
		return 0
	}
	var res []int = make([]int, len(triangle), len(triangle))
	copy(res, triangle[len(triangle) - 1])
	for n := len(triangle) - 1; n > 0; n-- {
		for j := 0; j < n; j++ {
			if res[j] < res[j + 1] {
				res[j] = res[j] + triangle[n - 1][j]
			} else {
				res[j] = res[j + 1] + triangle[n - 1][j]
			}
		}
	}
	return res[0]
}